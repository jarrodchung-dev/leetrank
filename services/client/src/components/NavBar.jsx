import React, { useEffect, useRef, useState } from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";

const NavBar = (props) => (
  // eslint-disable-next-line
  <header>
    <nav
      className="navbar is-mobile is-light"
      role="navigation"
      aria-label="main navigation"
    >
      <div className="navbar-brand">
        <button className="button is-light">
          <div className="navbar-item">
            <Link to="/">
              <strong>{props.title}</strong>
            </Link>
          </div>
        </button>
      </div>
    </nav>
  </header>
);

NavBar.propTypes = {
  title: PropTypes.string.isRequired,
  isAuthenticated: PropTypes.bool.isRequired
};

export default NavBar;

/*
<div className="container is-info">
  <div className="navbar-brand">
    <Link to="/">
      <button className="button is-light">
        <strong className="navbar-item">{props.title}</strong>
      </button>
    </Link>
    <span
      className="navbar-burger nav-toggle"
      onClick={() => {
        let toggle = document.querySelector(".nav-toggle");
        let menu = document.querySelector(".navbar-menu");
        toggle.classList.toggle("is-active");
        menu.classList.toggle("is-active");
      }}
    >
      <span></span>
      <span></span>
      <span></span>
    </span>
  </div>
  <div className="navbar-menu">
    <div className="navbar-start">
      <Link to="/" className="navbar-item">
        Home
      </Link>
      <Link to="/about" className="navbar-item">
        About
      </Link>
      <Link to="/all-users" className="navbar-item">
        Users
      </Link>
      {props.isAuthenticated && (
        <Link to="/status" className="navbar-item">
          User Status
        </Link>
      )}
    </div>
    <a href="/swagger" className="navbar-item">
      Swagger
    </a>
    <div className="navbar-end">
      {!props.isAuthenticated && (
        <div className="navbar-item">
          <Link to="/register" className="button is-primary">
            Register
          </Link>
          &nbsp;
          <Link to="/login" className="button is-link">
            Log In
          </Link>
        </div>
      )}
      {props.isAuthenticated && (
        <Link to="/logout" className="navbar-item">
          Log Out
        </Link>
      )}
    </div>
  </div>
</div>
*/
