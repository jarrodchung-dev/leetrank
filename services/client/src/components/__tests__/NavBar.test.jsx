import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import NavBar from "../NavBar";
import { MemoryRouter as Router } from "react-router-dom";

const title = "Hello, World!";
const isAuthenticated = true;

describe("NavBar", () => {
  test("NavBar renders properly", () => {
    const wrapper = shallow(<NavBar title={title} isAuthenticated={isAuthenticated} />);
    const element = wrapper.find("strong");
    expect(element.length).toBe(1);
    expect(element.get(0).props.children).toBe(title);
    expect(element.get(1).props.children).toBe(isAuthenticated);
  });

  test("NavBar renders a snapshot properly", () => {
    const tree = renderer
      .create(
        <Router location="/">
          <NavBar title={title} isAuthenticated={isAuthenticated} />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
