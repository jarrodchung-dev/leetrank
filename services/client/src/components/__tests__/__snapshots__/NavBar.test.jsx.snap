// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar NavBar renders a snapshot properly 1`] = `
<header>
  <nav
    aria-label="main navigation"
    className="navbar is-mobile is-light"
    role="navigation"
  >
    <div
      className="navbar-brand"
    >
      <button
        className="button is-light"
      >
        <div
          className="navbar-item"
        >
          <strong>
            <a
              href="/"
              onClick={[Function]}
            >
              Hello, World!
            </a>
          </strong>
        </div>
      </button>
    </div>
  </nav>
</header>
`;
